### PR TITLE
BUGS-5816: Resolves issue where changing file content would not serve…

### PR DESCRIPTION
**Pull request by rwagner00**
… new files.

Contributors:
* [nsciacca](https://git.drupalcode.org/nsciacca)
  * Original PR: [drupal.org](https://git.drupalcode.org/project/pantheon_advanced_page_cache/-/merge_requests/2)
* [Ec1ipsis](https://git.drupalcode.org/Ec1ipsis)

### CODE REVIEW

**Comment by stevector** [comment]
Can we get a Behat test for this too? Like https://github.com/pantheon-systems/pantheon_advanced_page_cache/blob/2.x/tests/behat/features/huge_header_warning.feature

**Comment by pwtyler** [comment]
Always +1 on tests.

**Comment by rwagner00** [comment]
Adding behat tests encountered numerous issues and turned turned into a _thing_ well outside the original scope of the ticket. After discussion with Greg I created https://getpantheon.atlassian.net/browse/CMSP-956 to do that part of the work so this ticket can be completed and the update delivered to customers. I just need to back out my previous commits and add the logging back.

File: pantheon_advanced_page_cache.module [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+/**
+ * Implements hook_ENTITY_TYPE_update() for file entities.
+ *
+ * Clears edge cache by file url. If this is not done, changing file contents
+ * does not actually change the file that gets served until the cache times out.
+ */
+function pantheon_advanced_page_cache_file_update(EntityInterface $file) {
+  // Don't try this if we're not in a Pantheon environment.
+  if (function_exists('pantheon_clear_edge_paths')) {
```
Comment (jazzsequence) [inline pantheon_advanced_page_cache.module]: Use the "happy path" and bail early. No need to nest all the functionality inside a conditional.

```php
if (! function_exists('pantheon_clear_edge_paths')) {
    return;
}
```

File: pantheon_advanced_page_cache.module [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    // No matter what, the file's base URL needs to be cleared out .
```
Comment (jazzsequence) [inline pantheon_advanced_page_cache.module]:
```suggestion
    // No matter what, the file's base URL needs to be cleared out.
```

**Review by namespacebrian — APPROVED** [review]

File: pantheon_advanced_page_cache.module [hunk 3]
```diff
@@ -0,0 +1,48 @@
+<?php
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\image\Entity\ImageStyle;
```
Comment (jazzsequence) [inline pantheon_advanced_page_cache.module:4]: Is `image` (lowercase `i`) correct here?

**Review by jazzsequence — COMMENTED** [review]
One major suggestion, one minor and a question.

File: pantheon_advanced_page_cache.module (hunk 3 above)
Comment (namespacebrian) [reply pantheon_advanced_page_cache.module:4]: yes, of course. :)   why wouldn't it be...
https://api.drupal.org/api/drupal/core%21modules%21image%21src%21Entity%21ImageStyle.php/8.2.x

**Review by jazzsequence — APPROVED** [review]

**Pull request merged**